src/services/codefixes/fixImportNonExportedMember.ts

```diff
@@ -0,0 +1,62 @@
+/* @internal */
+namespace ts.codefix {
+    const fixId = "importNonExportedMember";
+    const errorCodes = [
+        Diagnostics.Module_0_declares_1_locally_but_it_is_not_exported.code
+    ];
+    registerCodeFix({
+        errorCodes,
+        getCodeActions(context) {
+            const { sourceFile } = context;
+            const info = getInfo(sourceFile, context, context.span.start);
+            if (!info) {
+                return undefined;
+            }
+            const changes = textChanges.ChangeTracker.with(context, t => doChange(t, info.originSourceFile, info.node));
+            return [createCodeFixAction(fixId, changes, [Diagnostics.Export_0_from_module_1, info.node.text, showModuleSpecifier(info.importDecl)], fixId, Diagnostics.Export_all_non_exported_member)];
+        },
+        fixIds: [fixId],
+        getAllCodeActions: context => codeFixAll(context, errorCodes, (changes, diag) => {
+            const info = getInfo(diag.file, context, diag.start);
+            if (info) doChange(changes, info.originSourceFile, info.node);
+        }),
+    });
+
+    interface Info {
+        readonly node: Identifier;
+        readonly importDecl: ImportDeclaration;
+        readonly originSourceFile: SourceFile
+    }
+
+    function getInfo(sourceFile: SourceFile, context: CodeFixContext | CodeFixAllContext, pos: number): Info | undefined {
+        const node = getTokenAtPosition(sourceFile, pos);
+        if (node && isIdentifier(node)) {
+            const importStart = getLineStartPositionForPosition(pos, sourceFile);
+            const importDecl = tryGetImportFrom(getTokenAtPosition(sourceFile, importStart));

```

Comment:
I think you can just walk up the tree to find the nearest `import`. I think there's a function like `findAncestor`

Reply:
yes, findAncestor is a better choice :)